<?xml version="1.0"?>
<launch>

  <arg name="quad_name"/>
  <arg name="world_name"/>
  <arg name="paused"/>
  <arg name="gui"/>
  <arg name="use_ground_truth"/>
  <arg name="use_mpc" default="false"/>
  <arg name="enable_command_feedthrough" default="false"/>
  <arg name="custom_models" default=""/>

  <arg name="mav_name"/>
  <arg name="model"/>
  <arg name="enable_logging"/>
  <arg name="enable_ground_truth"/>
  <arg name="log_file"/>

  <arg name="x_init"/>
  <arg name="y_init"/>

  <arg name="debug"/>
  <arg name="verbose"/>

  <!-- Gazebo stuff to spawn the world !-->
  <env name="GAZEBO_MODEL_PATH" 
      value="${GAZEBO_MODEL_PATH}:$(find rotors_gazebo)/models:$(arg custom_models)"/>
  <env name="GAZEBO_RESOURCE_PATH" 
      value="${GAZEBO_RESOURCE_PATH}:$(find rotors_gazebo)/models"/>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_name)" />
    <arg name="debug" value="$(arg debug)" />
    <arg name="paused" value="$(arg paused)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="verbose" value="$(arg verbose)"/>
  </include>

  <!-- RotorS stuff to spawn the quadrotor !-->
  <group ns="$(arg mav_name)">
    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="$(arg mav_name)" />
      <arg name="model" value="$(arg model)" />
      <arg name="enable_logging" value="$(arg enable_logging)" />
      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
      <arg name="log_file" value="$(arg log_file)"/>
      <arg name="x" value="$(arg x_init)" />
      <arg name="y" value="$(arg y_init)" />
    </include>
  </group>

  <!-- RPG stuff !-->
  <group ns="$(arg quad_name)" >


      <node pkg="rpg_rotors_interface" type="rpg_rotors_interface" 
          name="rpg_rotors_interface" output="screen" >
        <rosparam file="$(find rpg_rotors_interface)/parameters/rpg_rotors_interface.yaml" />
        
        <remap from="odometry" to="ground_truth/odometry" />
        <remap from="rpg_rotors_interface/arm" to="bridge/arm" />
      </node>




  </group> 

</launch>
